test/e2e/pods.go

```diff
@@ -659,6 +659,364 @@ var _ = framework.KubeDescribe("Pods", func() {
 		})
 	})
 
+	It("should invoke init containers on a RestartNever pod [Conformance]", func() {
+		podClient := f.Client.Pods(f.Namespace.Name)
+
+		By("creating the pod")
+		name := "pod-init-" + string(util.NewUUID())
+		value := strconv.Itoa(time.Now().Nanosecond())
+		pod := &api.Pod{
+			ObjectMeta: api.ObjectMeta{
+				Name: name,
+				Labels: map[string]string{
+					"name": "foo",
+					"time": value,
+				},
+			},
+			Spec: api.PodSpec{
+				RestartPolicy: api.RestartPolicyNever,
+				InitContainers: []api.Container{
+					{
+						Name:    "init1",
+						Image:   "gcr.io/google_containers/busybox:1.24",
+						Command: []string{"/bin/true"},
+					},
+					{
+						Name:    "init2",
+						Image:   "gcr.io/google_containers/busybox:1.24",
+						Command: []string{"/bin/true"},
+					},
+				},
+				Containers: []api.Container{
+					{
+						Name:    "run1",
+						Image:   "gcr.io/google_containers/busybox:1.24",
+						Command: []string{"/bin/true"},
+					},
+				},
+			},
+		}
+		defer podClient.Delete(pod.Name, nil)
+		startedPod, err := podClient.Create(pod)
+		if err != nil {
+			framework.Failf("Error creating a pod: %v", err)
+		}
+		w, err := podClient.Watch(api.SingleObject(startedPod.ObjectMeta))
+		if err != nil {
+			framework.Failf("Error watching a pod: %v", err)
+		}
+		wr := watch.NewRecorder(w)
+		event, err := watch.Until(framework.PodStartTimeout, wr, client.PodCompleted)
+		Expect(err).To(BeNil())
+		framework.CheckInvariants(wr.Events(), framework.ContainerInitInvariant)
+		endPod := event.Object.(*api.Pod)
+
+		Expect(endPod.Status.Phase).To(Equal(api.PodSucceeded))
+		init := api.GetPodCondition(endPod.Status, api.PodInitialized)
+		Expect(init).NotTo(BeNil())
+		Expect(init.Status).To(Equal(api.ConditionTrue))
+
+		Expect(len(endPod.Status.InitContainerStatuses)).To(Equal(2))
+		for _, status := range endPod.Status.InitContainerStatuses {
+			Expect(status.Ready).To(BeTrue())
+			Expect(status.State.Terminated).NotTo(BeNil())
+			Expect(status.State.Terminated.ExitCode).To(BeZero())
+		}
+	})
+
+	It("should invoke init containers on a RestartAlways pod [Conformance]", func() {
```

Comment:
Will remove for now.

On Wed, May 4, 2016 at 10:22 PM, Prashanth B <notifications@github.com>
wrote:

> In test/e2e/pods.go
> <https://github.com/kubernetes/kubernetes/pull/23567#discussion_r62142629>
> :
>
> > +		endPod := event.Object.(*api.Pod)
> > +
> > +		Expect(endPod.Status.Phase).To(Equal(api.PodSucceeded))
> > +		init := api.GetPodCondition(endPod.Status, api.PodInitialized)
> > +		Expect(init).NotTo(BeNil())
> > +		Expect(init.Status).To(Equal(api.ConditionTrue))
> > +
> > +		Expect(len(endPod.Status.InitContainerStatuses)).To(Equal(2))
> > +		for _, status := range endPod.Status.InitContainerStatuses {
> > +			Expect(status.Ready).To(BeTrue())
> > +			Expect(status.State.Terminated).NotTo(BeNil())
> > +			Expect(status.State.Terminated.ExitCode).To(BeZero())
> > +		}
> > +	})
> > +
> > +	It("should invoke init containers on a RestartAlways pod [Conformance]", func() {
>
> conformance for an alpha feature? (I don't know what qualifies, feels
> weird especially since it's only implemented in docker)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/23567/files/50bdb2cadf45eb3bc2ca556ff91c3ab44dbaf1c3..5cac2f766c7f9473edee80cdee2f3d23c76c2eb1#r62142629>
>